File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.py
def solution(n):
    # 현재 숫자의 1의 개수를 세기
    target_cnt = bin(n).count('1')
    # 숫자를 1 늘리고 2진수로 변환해서 1의 개수가 같다면 해당 십진수 반환
    while True:
        n = n+1
        cnt = bin(n).count('1')
        if cnt == target_cnt:
            return n

File: 프로그래머스/2/12911. 다음 큰 숫자/다음 큰 숫자.java
class Solution {
    public static void log(Object o){
        System.out.println(o);
    }
    public static boolean isNext(int num, int target){
        StringBuilder sb = new StringBuilder();
        int cnt=0;
        while(num > 0){
            int remainder = num % 2;
            if(remainder == 1){
                cnt++;
            }
            if(cnt > target){
                return false;
            }
            sb.append(remainder);
            num /= 2;
        }
        return cnt == target;
    }
    public int solution(int n) {
        /*
        개수 세고
        해당 0과 1의 개수로 만들 수 있는 모든 수 구하기
        */
        
        // 개수 세기
        int targetCnt = 0;
        String nString = Integer.toString(n,2);
        for(int i=nString.length()-1; i >= 0; i--){
            char ch = nString.charAt(i);
            if(ch == '1'){
                targetCnt++;
            }
        }
        
        while(true){
            if(isNext(++n, targetCnt)){
                return n;
            }
        }
    }
}